tests/integration-tests/conftest.py

```diff
@@ -312,6 +313,60 @@ def _cluster_factory(cluster_config, extra_args=None, raise_on_error=True):
         factory.destroy_all_clusters(test_passed=request.node.rep_call.passed)
 
 
+@pytest.fixture(scope='class')
+def api_server_factory(cfn_stacks_factory, request, public_ecr_image_uri, api_definition_s3_uri):
```

Comment:
are we sure the api_servers are reused across tests? isn't the class scoped fixture recreated and so the reference to the existing api_servers lost?

Reply:
Converted to session based fixtures and tested by adding two tests to simply list clusters, the second test used a cached instance of the server (did not re-create the stack) then the stack was deleted at the end of the tests.